kernel/kernel-api/src/main/java/io/delta/kernel/internal/checksum/CachedCrcInfoResult.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.checksum;
+
+import io.delta.kernel.internal.lang.Lazy;
+import java.util.Optional;
+
+/**
+ * Represents the result of attempting to read and cache CRC information. This is a simple wrapper
+ * around Optional<Optional<CRCInfo>> to make the three-state nature more explicit and easier to
+ * work with.
+ *
+ * <p>The three states are:
+ *
+ * <ul>
+ *   <li>Not attempted - CRC reading was never tried
+ *   <li>Attempted with failure - CRC reading was tried but failed (parse error, I/O error, etc.)
+ *   <li>Attempted with success - CRC was successfully read and parsed
+ * </ul>
+ */
+public class CachedCrcInfoResult {
```

Comment:
The alternative to this class is we have `Optional<Optional<CRCInfo>>` in several different places, each place having to re-document what exactly are the three possible states. That's ugly and annoying. This is cleaner.

No need to make this more extensible yet to other cached-things, that would be premature optimization.